### DISCUSSION TITLE
How to include a code block in your comment

### DISCUSSION THREAD

**Discussion by becky-gilbert** [jspsych/jsPsych#1113] [General]
When posting a question on the forum, it's often helpful if you include your code. You can create a nicely-formatted code block in a discussion comment using three backticks (`` ` ``) before and after your code, like this:

````
```
<paste your code here>
```
````
And you can include the name of the programming language after the first set of backticks to get syntax highlighting. For instance, if you write the word javascript after the first three backticks, that will give you JavaScript syntax highlighting.
````javascript
```javascript
var foo = bar;
function say_hello() {
  console.log('hello');
}
```
````
Or just "js":
````js
```js
console.log('This works too!');
```
````
Same goes for HTML:
````html
```html
<div><p>Hello, world!</p></div>
```
````
After writing your code block, you can click on the 'Preview' tab at the top of the comment box to see how it will look when posted. 

See [here](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) for more tips on using markdown in Github comments. 😃

**Comment by jsdanie1** [comment]
Why when you click the <> Add Code Block, In the GITHub comment block. it doesn't default to this automatically. It only outputs `` which makes the code look horrible, Had to search this to finally get it to look right. Is this a GITHUB Bug?

**Reply by jsdanie1** [reply]
Apparently Slash Commands has this. I'll ignore the <> button going forward and just user that.

**Comment by ucycg** [comment]
What do I have to type to have highlighting for C language?

**Reply by jodeleeuw** [reply]
````c
```c
// this is C code
int x = 0;
```
````

**Reply by ucycg** [reply]
ty for your reply for some reason this wasn't working for me in the first place

**Comment by ronaldngounou** [comment]
How to add Github code like this when commenting?

<img width="849" height="358" alt="image" src="https://github.com/user-attachments/assets/0293f643-cb7c-4b45-849f-f3bcd65e8130" />

I have tried different ways like copy pasting the link but it doesn't render. What do you do in those cases?

**Reply by glav-git** [reply]
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet
